docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,179 @@
+
+
+## 3 Possible Solutions
+We have several possible options we can use to resolve this:
+ - Eliminate ML.NET native components and implement all functionality in managed code.
+ - Keep ML.NET native components and rewrite them to a avoid problematic dependencies completely.
+ - Keep ML.NET native components and ifdef/rewrite to avoid problematic dependencies on platforms/architectures only where they are not supported.
+ - Hybrid approach of replacement code and software fallback. This is the approach I recommend.
+
+None of these approaches resolve the 3rd party dependency issues. These solutions only deal with first party ML.NET code itself.
+
+I have lots more info about our dependency on x86/x64 in another document if required.
+
+### 3.1 Eliminate native components
+This is the most complicated solution and provides the least amount of short term benefit. Since x86/x64 run fine and gain performance benefits with these components, it doesn't make sense to spend the time to fully remove it. This is a possible solution, but not one that I would recommend, so I am not going to give more details on it unless we explicitly decide to go this route.
+
+### 3.2 Rewrite native components to work on other platforms
+This will still allow us to gain the benefits of the X86/x64 SIMD instructions and Intel MKL on architectures that support it but will also keep the benefits of native code in the other places. The downside is that we would have to build the native code for, potentially, a lot of different architectures.
+
+At a high level, this solution would require us to:
+ - Fix the build so it's not hardcoded to look for specific native dependencies.
+ - Fix the managed code so that if it can't find the native components the code behaves correctly.
+ - All the native code would need to have the x86/x64 dependencies change to other libraries or be re-written by us.
+   - CpuMathNative would need to be re-written to not use x86/x64 dependencies.
+   - FastTreeNative would need to be re-written to not use x86/x64 dependencies.
+   - LdaNative builds just find without x86/x64 dependencies, so no replacements would need to be found.
+   - MatrixFactorizationNative builds with the "USESSE" flag that requires x86/x64. We can conditionally enable/disable that flag and no other work would be required.
+   - MklProxyNative is only a wrapper for Intel MKL and can be ignored on non x86/x64 platforms. We will need to modify the build to exclude it as needed.
+   - SymSgdNative would need to be re-written to not use Intel MKL. There are only 4 methods that would need to be changed.
+
+I was unable to find all the replacements we would need. We would end up having to write many native methods ourselves with this approach.
+
+### 3.3 Rewrite native components to be only managed code
```

Comment:
What's the difference between 3.1 (eliminate native components) and 3.3 (rewrite native components to be only managed code)?

Reply:
So 3.1 was written back when I was thinking more about Intel MKL and not the native code as a whole. For 3.1 I was originally meaning fully remove the dependency we have on Intel MKL, but thats not applicable anymore. Your idea with 3.3 is what I was trying to convey with 3.4. I will re-word this section. 3.1 is going to be fully removed, and I will try and make the other differences more apparent. 

I think if we redid everything with hardware intrinsics we probably wouldn't need to use the native components, but I haven't done any perf testing myself to confirm that. I can run some preliminary tests with CPUMath since we have native and intrinsics for that.